stack/templates/deployment.yaml

```
@@ -71,9 +71,15 @@ spec:
           {{- end }}
           resources:
             {{- toYaml .Values.resources | nindent 12 }}
-          {{- with .Values.volumeMounts }}
+          {{- if or (and .Values.persistence.enabled .Values.persistence.mountPath) .Values.volumeMounts}}
           volumeMounts:
-            {{- toYaml . | nindent 12 }}
+            {{- if and .Values.persistence.enabled .Values.persistence.mountPath) }}
```

Oops, my previous code suggestion wasn't quite right
```suggestion
            {{- if and .Values.persistence.enabled .Values.persistence.mountPath }}
```